bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -759,39 +764,44 @@ async fn handle_event_bus_event(
 }
 
 async fn handle_next_invocation(
-    next_response: Result<NextEventResponse>,
+    client: &Client,
+    ext_id: &str,
     invocation_processor: Arc<TokioMutex<InvocationProcessor>>,
-) -> bool {
+) -> NextEventResponse {
+    let next_response = next_event(client, ext_id).await;
     match next_response {
         Ok(NextEventResponse::Invoke {
-            request_id,
+            ref request_id,
```

Comment:
Maybe the `ref` is not required for `request_id`?

Reply:
<img width="939" alt="image" src="https://github.com/user-attachments/assets/93fe0f27-7ac8-431e-a336-d8485544033f" />
Nah that's a partial move